pages/api/auth/[...solana].ts

```diff
@@ -0,0 +1,5 @@
+import { FirebaseAdapter } from "../../../utils/adapters/firebase";
```

Comment:
If someone uses nextauth on the same project, this is going to cause conflicts

I'd change the route here to pages/api/solana-auth/[...method].ts

-----

Also, if we don't plan to have nested routes (e.g. /solana-auth/nested/route), we can make this less error prone by replacing it with

`pages/api/solana-auth/[method].ts`